### PR TITLE
fetch post

### PR DESCRIPTION
**Pull request by DominikReszczynski**

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
It should save new flashcard to the database when adding one

### Requirements
1. Fetch API must be used
2. Following endpoint must be used: 
```bash
curl -X POST \
https://training.nerdbord.io/api/v1/fischkapp/flashcards \
-H 'Authorization: "secret_token" \
-H 'Content-Type: application/json' \
-d '{
  "front": "What is life?",
  "back": "I have no idea.",
}'
```

### Code Review

#### src/App.tsx
- The `URL_TOKEN` has been updated to 'secret_token' which matches the task requirement.
- An `id` field has been added to the `Flashcard` interface and is being used when mapping the data received from the server. This does not directly relate to the task requirements, but it does not contradict them either.
- The server response is being logged to the console. This is not mentioned in the task requirements, but it can be useful for debugging purposes.

#### src/components/AppendCard/AppCardAdd.tsx
- The Fetch API is being used to send a POST request to the specified endpoint, which matches the task requirements.
- The 'Authorization' and 'Content-Type' headers are being set correctly as per the task requirements.
- The body of the request is a hardcoded object with `front` and `back` fields. This does not match the task requirements, as the values should be taken from the `firstWord` and `secondWord` states, not hardcoded.
- The server response is being logged to the console. This is not mentioned in the task requirements, but it can be useful for debugging purposes.
- There is a typo in 'Content-Type' header value. It should be "application/json" instead of "pplication/json".

### Conclusion
The code changes partially cover the task requirements. The Fetch API is being used to send a POST request to the correct endpoint with the correct headers. However, the body of the request is not being constructed correctly. The `front` and `back` fields should be taken from the `firstWord` and `secondWord` states, not hardcoded. Also, there is a typo in 'Content-Type' header value. It should be "application/json" instead of "pplication/json".

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task Goal
The task goal is to save a new flashcard to the database when adding one.

### Requirements
1. Fetch API must be used
2. The following endpoint must be used: 
```bash
curl -X POST \
  https://training.nerdbord.io/api/v1/fischkapp/flashcards \
  -H 'Authorization: "secret_token" \
  -H 'Content-Type: application/json' \
  -d '{
    "front": "What is life?",
    "back": "I have no idea.",
}'
```

### Code Changes Review

#### File: src/App.tsx
The changes in this file are not directly related to the task requirements. However, the changes are correct and do not introduce any bugs. The `URL_TOKEN` has been updated to match the task requirements and an `id` field has been added to the `Flashcard` interface and used when mapping the data received from the server.

#### File: src/components/AppendCard/AppCardAdd.tsx
The changes in this file are directly related to the task requirements and they correctly implement the requirements. The `fetch` API is used to send a POST request to the specified endpoint with the correct headers and body. The body of the request is correctly formatted as JSON and includes the `front` and `back` fields as required.

However, there is a minor issue in the way the `data` object is created. The `front` and `back` fields are set to objects instead of strings. This could cause issues when the server tries to parse the request body. The correct code should be:
```javascript
const data = {
  front: firstWord,
  back: secondWord
};
```

Overall, the code changes cover the task requirements with a minor issue that needs to be addressed.

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
It should save new flashcard to the database when adding one.

### Requirements
1. Fetch API must be used
2. Following endpoint must be used: 
```bash
curl -X POST \
  https://training.nerdbord.io/api/v1/fischkapp/flashcards \
  -H 'Authorization: "secret_token" \
  -H 'Content-Type: application/json' \
  -d '{
    "front": "What is life?",
    "back": "I have no idea.",
}'
```

### Code Changes Review

#### File: src/App.tsx
The changes in this file are mostly about adding an `id` field to the Flashcard interface and updating the `URL_TOKEN` constant. These changes are not directly related to the task requirements but they do not contradict them either.

#### File: src/components/AppendCard/AppCardAdd.tsx
The changes in this file are directly related to the task requirements. The `addCard` function now uses the Fetch API to send a POST request to the specified endpoint with the required headers and body. This is in line with the task requirements.

However, there is a potential issue with this code. The `addCard` function sends the request but it does not update the `cards` state with the new card. According to the task goal, the new flashcard should be saved to the database when adding one. This implies that the local state should also be updated to reflect this change. This is not currently happening in the provided code changes.

In conclusion, the code changes partially cover the task requirements. The Fetch API is used correctly and the correct endpoint is used. However, the local state is not updated after adding a new card, which may not fully meet the task goal.

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task goal
The task goal was to save a new flashcard to the database when adding one.

### Requirements
1. Fetch API must be used
2. Following endpoint must be used: `https://training.nerdbord.io/api/v1/fischkapp/flashcards`

### Code Changes Review

#### src/App.tsx
The changes in this file seem to align with the task requirements. The `URL_TOKEN` has been updated to 'secret_token' as per the requirements. A new state variable `post` has been added and used in the `useEffect` hook to trigger a re-fetch whenever a new card is posted. The `setCards` function is now being called with an empty array before setting it with the new cards, which seems unnecessary but doesn't affect functionality. The `onAddCard` prop passed to `AppCardAdd` has been split into `onAddCardTrue` and `onAddCardFalse`, which seems to be used for controlling the visibility of the add card form.

#### src/components/AppCard/AppCard.tsx
The changes in this file are minor and seem to be more about fixing previous issues rather than implementing the task requirements. The `front` and `back` props passed to `NormalCard` and `EditCard` are no longer being accessed as `front.firstWord` and `back.secondWord`, but directly as `front` and `back`. This change seems to be in line with the changes in the other files where the structure of a card has been simplified.

#### src/components/AppendCard/AppCardAdd.tsx
The changes in this file cover the task requirements. A fetch POST request is being made to the specified URL with the correct headers and body format. The `onAddCard` prop has been split into `onAddCardTrue` and `onAddCardFalse`, which are used for controlling the visibility of the add card form. The `addCard` function no longer updates the local state with the new card, but instead sends a POST request to the server.

### Conclusion
The code changes cover the task requirements. The new flashcard is being saved to the database using a fetch POST request to the specified endpoint. The only potential issue is that the local state is not being updated with the new card after it's posted, but this is mitigated by the re-fetch triggered in `App.tsx`.
